classes/Deck.js

```diff
@@ -1,22 +1,187 @@
-import { PlayingCard } from './PlayingCard'
+import { SUITES, RANKS } from '../constants/playing-card'
 
 class Deck {
-  constructor(suites, ranks) {
-    const cards = []
-    for (const suite of suites) {
-      for (const rank of ranks) {
-        cards.push(new PlayingCard(suite, rank))
+  constructor(cards = []) {
+    this.cards = cards
+  }
+
+  validateCanDrawCard(cardPosition) {
+    if ((0 >= cardPosition) || (cardPosition > this.size())) {
+      throw new Error(`RangeError: Cannot remove card from position (${cardPosition})`)
+    }
+  }
+
+  validateCanInsertCard(cardPosition) {
+    if ((0 >= cardPosition) || (cardPosition > this.size() + 1)) {
+      throw new Error(`RangeError: Cannot insert card to position ${cardPosition}`)
+    }
+  }
+
+  validateNotEmpty() {
+    if (this.isEmpty()) {
+      throw new Error(`IllegalOperation: Cannot draw card, deck is empty`)
+    }
+  }
+
+  validateNotFull() {
+    if (this.isFull()) {
+      throw new Error(`IllegalOperation: Cannot insert card, deck is full`)
+    }
+  }
+
+  print() {
+    this.cards.forEach(console.log)
+  }
+
+  drawCard(cardPosition) {
+    this.validateNotEmpty()
+    this.validateCanDrawCard(cardPosition)
+
+    return this.cards.splice(cardPosition - 1, 1)[0]
```

Comment:
Would probably be better to use class methods here. It should very similar to how you've implemented `insertCard`, only removing a card instead of adding one